add_codeowners.py

```diff
@@ -0,0 +1,498 @@
+    base_branch_name: str = "master",
+    title: str = "AUTOGENERATED: Pull Request Title",
+    body: str = "(autogenerated) Pull Request body",
+    api: Optional[GhApi] = None,
+) -> None:
+    """
+    Create a pull request to merge:
+    `contributor`/`branch_name` --> `upstream`/`base_branch_name`
+    """
+    if api is None:
+        api = GhApi()  # requires a token for access
+
+    # Create pull request
+    print(f" > Creating pull request: {title}")
+    api.pulls.create(
+        owner=contributor,
+        repo=repo_name,
+        title=title,
+        head=f"{upstream}:{branch_name}",  # This could also point to contributor
+        base=base_branch_name,
+        body=body,
+        maintainer_can_modify=True
+    )
+
+
+def add_codeowners_from_setting(
+    settings: RepoOwnerSettings,
+    user: str,
+    branch_name: str,
+    commit_message: str,
+    pr_title: str,
+    pr_body: str,
+    api: Optional[GhApi],
+):
+    """
+    Add a CODEOWNERS file from the specified `RepoOwnerSettings` object.
+
+    Also requires information about the actor (`user`), and the contribution
+    description metadata (branch_name, commit_message, pr_title, pr_body)
+    """
+    if api is None:
+        api = GhApi()  # requires a token for access
+
+    # create_fork(
+    #     repo_name=settings.repo_name,
+    #     original_owner=settings.owner,
+    #     api=api,
+    # )
+    repo_info = api.repos.get(settings.owner, settings.repo_name)
+    default_branch_name = repo_info["default_branch"]
+    create_branch(
+        owner=settings.owner,
+        repo_name=settings.repo_name,
+        branch_name=branch_name,
+        default_branch_name=default_branch_name,
+        api=api,
+    )
+    codeowner_file = create_codeowners_file(settings)
+    print("")
+    print(base64.b64decode(codeowner_file).decode('utf-8'))
+    add_codeowners_file_to_branch(
+        owner=settings.owner,
+        repo_name=settings.repo_name,
+        branch_name=branch_name,
+        codeowner_data=codeowner_file,
+        commit_message=commit_message,
+        api=api,
+    )
+    create_pull_request(
+        repo_name=settings.repo_name,
+        contributor=user,
+        upstream=settings.owner,
+        base_branch_name=default_branch_name,
+        branch_name=branch_name,
+        title=pr_title,
+        body=pr_body,
+        api=api,
+    )
+
+
+def main(
+    owner: str = "pcdshub",
+    user: str = "pcdshub",
+    repo_name: str = "",
+    branch_name: str = "mnt_add_codeowners",
+    commit_message: str = "MNT: Adding CODEOWNERS file",
+    title="MNT: Adding CODEOWNERS file",
+    body="(autogenerated) adding codeowners file",
+    sme: Optional[List[str]] = None,
+    area: Optional[List[str]] = None,
+    lang: Optional[List[str]] = None,
+    repo_data_path: str = "",
+    write: bool = False
+):
+    """
+    Create codeowners files and submit them via pull request
+    """
+    # Set dry-run option
+    DryRunner.set(run=write)
```

Comment:
Comment:
I'm a big fan of this, I'm interested in using this or something like it in future scripts